### PR TITLE
Removed quick_start parameter on plm and exposed load_aldb

### PR DESCRIPTION
**Pull request by wz2b**
I removed the quick_start parameter on plm, and replaced it with load_aldb which was present at the PLM layer but not at Connection (where one creates a PLM).  This exposed what was essentially the 'quick start' feature that doesn't appear to have been completed.

### CODE REVIEW

File: insteonplm/plm.py [hunk 1]
```diff
@@ -343,7 +342,7 @@ def _setup_devices(self):
                       len(self.devices.saved_devices))
         self._get_plm_info()
         self.devices.add_known_devices(self)
-        if self._quick_start:
+        if self._load_aldb == False:
```
Comment (teharris1) [inline insteonplm/plm.py]: Change the order of this: 
If self._load_aldb:
    self._load_all_link_database()
else:
    self._complete_setup()

**Comment by teharris1** [comment]
Thanks for looking into this. I saw your issues come in but have not had time to look at them.

**Comment by wz2b** [comment]
I'm happy to help - I want to be a contributor, not just a complainer.  I
don't know anything about flake8 or pylint (other than what they are) but
I'll look into it.

When I fix that if then else ordering do I submit a second pull request?
Or can I somehow modify the current request?

**Comment by wz2b** [comment]
Since this is the first time I'm doing this I need a walk-through on a few pieces:

```************* Module insteonplm
insteonplm/__init__.py:135:4: R0913: Too many arguments (11/10) (too-many-arguments)

------------------------------------
Your code has been rated at 10.00/10

ERROR: InvocationError for command '/home/chrisp/work/python-insteonplm/.tox/pylint/bin/pylint insteonplm tests' (exited with code 8)
```

The too many arguments is because pylint doesn't want more than 10, as I mentioned in the previous comment I'm not sure I can do anything about that without either more major changes, or risking breaking backward compatibility.  But that InvocationError is something I couldn't figure out how to fix.  pylint is there and the stated location, and it runs.  Is this the problem?

```SKIPPED: InterpreterNotFound: python3.6```

**Comment by teharris1** [comment]
When you commit your changes to your forked repository this PR will update with the new code. As for the lint issue of 'too-many-arguments' it is OK to explicitly tell the interpreter to ignore that issue with the following tag just above the method definition:
`# pylint: disable=too-many-arguments`
This allows us to find those later and fix them when it makes sense. But for now it just a matter of documenting them. At some point we will need to make breaking changes but not for each PR. 

tox is the right way to run the tests since it also runs the `pytest` scripts as well. You can also link your repository to `travis-ci` which will run the tests each time you commit.

**Comment by wz2b** [comment]
>  When you commit your changes to your forked repository this PR will
update with the new code.

Organizationally, does that mean I screwed up, and I shouldn't be issuing a
PR on Master, but rather I should have created a feature branch or possibly
an 'upstream to be committed' branch?

**Comment by teharris1** [comment]
Just to give you a little more background on the "fast start" option I was thinking about, there are a few concepts I was playing with but they were not high enough priority for me to finish the code. Here they are:
1) The library has to have a device in the `IM.devices` list in order to do anything.  There are three ways to load that list.
  a. Load the ALDB and perform the `IM.get_device_info` process to identify the devices (this is the default way)
  b. Load the saved devices which already has the most recently known devices
  c. "Manually" add the devices based on some other external list.
2.  Option "c" has a lot of issues with it but probably still needs to be there. The issue is it runs the risk of a device being added that is not in the IM's ALDB. In that situation, the IM and the device communications will produce flaky results. (one is talking but the other is not listening.)
3. If you load devices with option "b" that is very fast and no traffic. It is also reliable if it was generated via option "a".
4. Option "a" is very reliable (except for batter operated devices that don't wake up, but that is why "device override" was created.). But it takes much longer (i.e. 5  - 15 secs per device)

So with all that background, the question is 'if you have a saved device file, i.e. option "b", do you just use it and skip option "a"?' The implication of this is if a device was added to the IM's ALDB you may not know about it until you reload the ALDB. The current start up process loads the saved devices then loads the ALDB, this was a trade off to allow the devices to load quickly but also to ensure that new devices were found. There are two downsides to this:
1. After the devices are loaded there is a lot of traffic generated so the devices are useless anyway because the messages are sent in a serial mode. In other words, the devices cannot be manipulated until the ALDB laod process finishes anyway.
2. If you remove a device from the IM's ALDB offline using another tool the device still shows up. Not as concerned about that one since this is outside our control and is fixed by simply deleting the `insteon_plm_device_info.dat` file. 

The conclusion I came to is:
1. Load the ALDB if there is no saved device file
2. Allow the ALDB to be purged and reloaded so the user can "refresh" the ALDB when they make changes. this will then need to refresh the saved device file.
3. Ensure that the All-Link process updates the ALDB so that if a device is added or removed via this library the IM's ALDB is in sync with that change. This will also need to refresh the saved device file.

I am not opposed to this PR since it does not break anything and allows a consumer of the library to decide how they want to populate `IM.devices`. So no worries there but wanted to give you a full picture of my thought process.

**Comment by teharris1** [comment]
> Organizationally, does that mean I screwed up, and I shouldn't be issuing a
PR on Master, but rather I should have created a feature branch or possibly
an 'upstream to be committed' branch?

Technically, no but now your repository has commits that will not really be in this `master` repository. The issue is more on your side now. Every time you post a new PR you will be reposting all of your prior commits unless you blow away your fork and start from a new clone. So the best practice for you is:
1) Fork this repository
2) Create a new branch
3) COmmit to the new branch and submit a PR agains this `master` branch from that branch
4) After the PR is approved/rejected, delete that branch
5) Go back to your `master` branch and merge with the upstream (i.e. this repository) `master` branch.

What this does is it allows you to have a forked `master` branch that always tracks this `master`.  All your feature branches will come from your `master` branch. This way you don't have to blow away and recreate your repository each time you post a PR. Remember to merge your forked `master` with this repository every time you start a new feasture that way you are working against the most recent code base.

**Pull request merged**